public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/SmartAlertTypeDetector.tsx

```diff
@@ -0,0 +1,170 @@
+    !editingExistingRule &&
+    !isRecordingRuleType &&
+    ((cloudTypeEnabled && canBeCloud && ruleFormType === RuleFormType.grafana) ||
+      (ruleFormType === RuleFormType.cloudAlerting && grafanaTypeEnabled));
+  return canSwitch;
+};
+
+export function SmartAlertTypeDetector({
+  editingExistingRule,
+  rulesSourcesWithRuler,
+  queries,
+  removeExpressionsInQueriesReducer,
+  addExpressionsInQueries,
+  prevExpressions,
+  setPrevExpressions,
+}: {
+  editingExistingRule: boolean;
+  rulesSourcesWithRuler: Array<DataSourceInstanceSettings<DataSourceJsonData>>;
+  queries: AlertQuery[];
+  removeExpressionsInQueriesReducer: () => void;
+  addExpressionsInQueries: (expressions: AlertQuery[]) => void;
+  prevExpressions: AlertQuery[];
+  setPrevExpressions: (expressions: AlertQuery[]) => void;
+}) {
+  const { getValues, setValue } = useFormContext<RuleFormValues>();
+
+  const [ruleFormType, dataSourceName] = getValues(['type', 'dataSourceName']);
+  const styles = useStyles2(getStyles);
+
+  const canSwitch = getCanSwitch({ queries, ruleFormType, editingExistingRule, rulesSourcesWithRuler });
+
+  const restoreExpressionsInQueries = useCallback(() => {
+    addExpressionsInQueries(prevExpressions);
+  }, [prevExpressions, addExpressionsInQueries]);
+
+  const onClickSwitch = useCallback(() => {
+    const typeInForm = getValues('type');
+    if (typeInForm === RuleFormType.cloudAlerting) {
+      setValue('type', RuleFormType.grafana);
+      setPrevExpressions.length > 0 && restoreExpressionsInQueries();
+    } else {
+      setValue('type', RuleFormType.cloudAlerting);
+      const expressions = queries.filter((query) => query.datasourceUid === ExpressionDatasourceUID);
+      setPrevExpressions(expressions);
+      removeExpressionsInQueriesReducer();
+    }
+  }, [
+    getValues,
+    setValue,
+    queries,
+    removeExpressionsInQueriesReducer,
+    restoreExpressionsInQueries,
+    setPrevExpressions,
+  ]);
+
+  const typeTitle =
+    ruleFormType === RuleFormType.cloudAlerting ? 'Data source-managed alert rule' : 'Grafana-managed alert rule';
+  const switchToLabel = ruleFormType !== RuleFormType.cloudAlerting ? 'data source-managed' : 'Grafana-managed';
+  const contentText =
+    ruleFormType === RuleFormType.cloudAlerting
+      ? 'Data source-managed alert rules are stored in the data source and are managed by the data source.'
+      : 'Grafana-managed alert rules are stored in the Grafana database and are managed by Grafana.';
+  const titleLabel =
+    ruleFormType === RuleFormType.cloudAlerting
+      ? `This rule is going to be managed by the '${dataSourceName}' data source. 
+      If your data source does not have an Alert manager configured or you wish to use expressions or multiple queries, switch to a Grafana-managed alert.`
+      : 'This is a Grafana-managed alert rule. You can switch it to a data source-managed alert rule if you have a Mimir / Loki / Cortex data source configured with an Alert manager.';
```

Comment:
Should we say in this text that expressions and multiple queries are not available in data source-managed rules?

Reply:
this information is in the Need Help info popup